src/fields/field2_128/mod.rs

```diff
@@ -25,13 +31,33 @@ use subtle::ConstantTimeEq;
 pub struct Field2_128(u128);
 
 impl Field2_128 {
+    const SUBFIELD_BIT_LENGTH: usize = 16;
+
     /// Project a u128 integer into a field element.
     ///
     /// This duplicates `FieldElement::from_u128()` in order to provide a const function with the
     /// same functionality, since trait methods cannot be used in const contexts yet.
     const fn from_u128_const(value: u128) -> Self {
         Self(value)
     }
+
+    /// Inject the value into the field using the subfield basis, per [2.2.2][1]. The basis only has
+    /// 16 elements, so we can't inject anything bigger than u16.
+    ///
+    /// [1]: https://datatracker.ietf.org/doc/html/draft-google-cfrg-libzk-01#section-2.2.2
+    // Allow dead_code for now. This will get used when we wire up proofs on MACs.
+    #[allow(dead_code)]
+    fn inject(mut value: u16) -> Self {
+        let mut injected = Self::ZERO;
+        for basis_element in subfield_basis() {
+            if value & 1 == 1 {
+                injected += basis_element;
+            }
```

Comment:
We should be doing this in constant time. We ought to implement `ConditionallySelectable` for all field elements to help with this sort of thing. I think it would make sense to take care of this in a follow-up PR.